File: app/src/main/java/com/ryanrvldo/moviecatalogue/ui/home/SplashScreenActivity.java
package com.ryanrvldo.moviecatalogue.ui.home;

import android.content.Intent;
import android.os.Bundle;
import android.os.Handler;

import androidx.appcompat.app.AppCompatActivity;
import androidx.navigation.ActivityNavigator;

import com.ryanrvldo.moviecatalogue.databinding.ActivitySplashScreenBinding;

public class SplashScreenActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        ActivitySplashScreenBinding binding = ActivitySplashScreenBinding.inflate(getLayoutInflater());
        setContentView(binding.getRoot());

        new Handler().postDelayed(() -> {
            ActivityNavigator navigator = new ActivityNavigator(this);
            navigator.navigate(navigator.createDestination().setIntent(new Intent(this, MainActivity.class)), null, null, null);
            finish();
        }, 2000);
    }
}

File: app/src/main/java/com/ryanrvldo/moviecatalogue/ui/SplashScreenActivity.kt
package com.ryanrvldo.moviecatalogue.ui

import android.content.Intent
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.lifecycle.lifecycleScope
import androidx.navigation.ActivityNavigator
import com.ryanrvldo.moviecatalogue.databinding.ActivitySplashScreenBinding
import kotlinx.coroutines.delay

class SplashScreenActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val binding = ActivitySplashScreenBinding.inflate(layoutInflater)
        setContentView(binding.root)

        launchMainActivity()
    }

    private fun launchMainActivity() = lifecycleScope.launchWhenCreated {
        delay(2000)
        val navigator = ActivityNavigator(this@SplashScreenActivity)
        navigator.navigate(
            navigator.createDestination()
                .setIntent(Intent(this@SplashScreenActivity, MainActivity::class.java)),
            null,
            null,
            null
        )
        finishAfterTransition()
    }
}